king_phisher/client/windows/campaign_import.py

```diff
@@ -0,0 +1,439 @@
+		campaign_name = self.campaign_info.find('name').text
+		campaign_check = self.rpc('db/table/get', 'campaigns', campaign_id)
+		if not campaign_check:
+			return
+		if campaign_name == campaign_check['name']:
+			self.rpc('db/table/delete', 'campaigns', campaign_id)
+			self.logger.info('Deleted Campaign {}'.format(campaign_id))
+
+	def signal_import_button(self, _):
+		"""
+		Function called when the user starts the import campaign process, It will create a new thread,
+		to have the campaign run in the background freeing up the GUI for the user to conduct other functions.
+		"""
+		if not self.campaign_info:
+			self._update_text_view('No campaign information to import')
+			self.button_import_campaign.set_sensitive(False)
+			return
+		self.thread_import_campaign = ImportThread(target=self._import_campaign)
+		self.thread_import_campaign.start()
+
+	def select_xml_campaign(self):
+		"""
+		Prompts the user with a file dialog window to select the King Phisher Campaign XML file to import.
+		Validates the file to make sure it is a Campaign exported from King Phisher and is the correct version to import.
+		"""
+		dialog = extras.FileChooserDialog('Import Campaign from XML', self.window)
+		dialog.quick_add_filter('King Phisher XML Campaign', '*.xml')
+		dialog.quick_add_filter('All Files', '*')
+		response = dialog.run_quick_open()
+		dialog.destroy()
+		if not response:
+			return
+		target_file = response['target_path']
+		self.entry_path.set_text(target_file)
+
+		try:
+			campaign_xml = ET.parse(target_file)
+		except ET.ParseError as error:
+			self.logger.error('cannot import campaign: {0} is not a valid XML file'.format(target_file), error)
+			raise KingPhisherInputValidationError('{0} is not a valid xml file'.format(target_file))
+
+		root = campaign_xml.getroot()
+		if root.tag != 'king_phisher':
+			raise KingPhisherInputValidationError('File not a King Phisher Campaign XML Export File')
+		meta_data = root.find('metadata')
+		if meta_data.find('version').text < '1.3':
+			raise KingPhisherInputValidationError('Can only import XML Campaign data version 1.3 or higher')
+		self.campaign_info = root.find('campaign')
+		if not self.campaign_info:
+			raise KingPhisherInputValidationError('XML File does not contain any campaign information')
+
+		self.db_campaigns = self.rpc.graphql("{ db { campaigns { edges { node { id, name } } } } }")['db']['campaigns']['edges']
+		self.entry_campaign_name.set_text(self.campaign_info.find('name').text)
+		self.thread_import_campaign = None
+		if not self._check_campaign_name(self.campaign_info.find('name').text, verbose=True):
+			self.button_import_campaign.set_sensitive(False)
+			return
+		self.button_import_campaign.set_sensitive(True)
+
+	def _check_campaign_name(self, campaign_name, verbose=False):
+		"""
+		Used to check to see the user has provided a campaign name that is not already in use.
+		If verbose is used it will display information in detailed output text buffer.
```

Comment:
When referencing a parameter within the functions documentation, I think you're supposed to use ```**``` so sphinx will adjust the formatting.